packages/editor/src/bindings/post-meta.js

```diff
@@ -7,22 +7,43 @@ import { store as coreDataStore } from '@wordpress/core-data';
  * Internal dependencies
  */
 import { store as editorStore } from '../store';
+import { unlock } from '../lock-unlock';
+
+function getMetadata( registry, context ) {
+	let metaFields = {};
+	const { type } = registry.select( editorStore ).getCurrentPost();
+	const { getEditedEntityRecord } = registry.select( coreDataStore );
+	const { getRegisteredPostMeta } = unlock(
+		registry.select( coreDataStore )
+	);
+
+	if ( type === 'wp_template' ) {
```

Comment:
What about `wp_block`, `wp_template_part` types? I don't think we should expose their meta fields to users.

What happens when there is a template with a block that changes the `postType` and `postId` through context, should it still be handled differently than outside of `wp_template`?

Reply:
If I am not mistaken, `wp_block` and `wp_template_part` would go through the other part of [this conditional](https://github.com/WordPress/gutenberg/blob/d724026a292a79c63c0beb6fdea701139ab38d5f/packages/editor/src/bindings/post-meta.js#L22-L38) which would return nothing.

However, I believe we can make it the other way around to cover the cases mentioned here with something like this:

```js
	if ( context?.postType && context?.postId ) {
		metaFields = getEditedEntityRecord(
			'postType',
			context?.postType,
			context?.postId
		).meta;
	} else if ( type === 'wp_template' ) {
		// Populate the `metaFields` object with the default values.
		Object.entries( registeredFields || {} ).forEach(
			( [ key, props ] ) => {
				if ( props.default ) {
					metaFields[ key ] = props.default;
				}
			}
		);
	}

```

Reply:
I have created [this follow-up pull request](https://github.com/WordPress/gutenberg/pull/65449) to handle this. Let's discuss there any potential change.